app/v2/notifications/post_notifications.py

```diff
@@ -334,13 +336,22 @@ def process_document_uploads(personalisation_data, service, simulated=False):
         service_id=authenticated_service.id
     )
 
+    error_if_service_using_email_verification_flow_without_permission(
+        {k: personalisation_data[k] for k in file_keys},
+        service.permissions
+    )
+
     for key in file_keys:
         if simulated:
             personalisation_data[key] = document_download_client.get_upload_url(service.id) + '/test-document'
         else:
+            verify_email = personalisation_data[key].get('verify_email_before_download') or False
```

Comment:
could we set this variable before calling `error_if_service_using_email_verification_flow_without_permission` and then pass it, and then `error_if_service_using_email_verification_flow_without_permission` could be simpler?

Reply:
Done 👍

Reply:
✨